src/hash/rpo/digest.rs

```diff
@@ -73,6 +79,29 @@ impl From<[Felt; DIGEST_SIZE]> for RpoDigest {
     }
 }
 
+impl From<[u64; DIGEST_SIZE]> for RpoDigest {
+    fn from(value: [u64; DIGEST_SIZE]) -> Self {
+        Self([
+            Felt::new(value[0]),
+            Felt::new(value[1]),
+            Felt::new(value[2]),
+            Felt::new(value[3]),
+        ])
+    }
+}
+
+impl From<[u8; DIGEST_BYTES]> for RpoDigest {
+    fn from(value: [u8; DIGEST_BYTES]) -> Self {
+        // Note: `unwrap`s below are safe since we know the length of the input
+        Self([
+            value[0..8].try_into().unwrap(),
+            value[8..16].try_into().unwrap(),
+            value[16..24].try_into().unwrap(),
+            value[24..32].try_into().unwrap(),
+        ])
+    }
+}
```

Comment:
Should we check if integers are within range for field elements and panic otherwise? Without this check, one potential issue could be that two different strings of bytes are deserialized as the same value. Shouldn't be an issue in most case, but maybe it is better to be on the save side.

Reply:
Oh! I didn't notice, but the `try_into` implementation used here was delegating to [this](https://github.com/facebook/winterfell/blob/7dbb608e343402defa6b01c2d893200a470a7776/math/src/field/f64/mod.rs#L566-L598) implementation, which does the validation you mentioned.

However, this is an attack vector, should we use this to parse data from user input, one can trigger a panic. I'm going to update this to use `try_into` and return an error instead, how does that sound?

Reply:
> now I removed the `From` trait implementations, the assumption is that it doesn't make sense to have user provided values (as in, the user chooses a hex string of they liking), only values that have been generated by the system (as in, the result of a hashing operation), so the even if the silent modulus conversion is possible, that would likely be hiding a bug.

Yes - I think this works!